### ISSUE TITLE
[Bug]: Argument shapes are inconsistent

### ISSUE THREAD

**Issue by jikechao** [openvinotoolkit/openvino#21247]
### OpenVINO Version

openvino-nightly 2023.2.0.dev20231101

### Operating System

Ubuntu 18.04 (LTS)

### Device used for inference

CPU

### Framework

Keras (TensorFlow 2)

### Model used

Given in the following script

### Issue description

OpenVINO converts the model with the operator `BatchNormalization` failed.

### Step-by-step reproduction

```
import tensorflow as tf
from tensorflow import keras as keras
from tensorflow.keras import layers, models
import numpy as np
import openvino as ov

layer = keras.layers.BatchNormalization()
input_shape = [1, 2, 3, 4, 5]
input_data = np.random.random(input_shape).astype(np.float32)

x = layers.Input(shape=input_shape[1:], dtype="float32")
y = layer(x)
model = models.Model(x, y)
# model.summary()
res_keras = model(input_data)

tf2_model_path = f"_temp_model"
tf.saved_model.save(model, tf2_model_path)
ov_model = ov.convert_model(tf2_model_path,  input=input_shape)  # crash here!
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "test.py", line 20, in <module>
    ov_model = ov.convert_model(tf2_model_path,  input=input_shape)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert.py", line 101, in convert_model
    ov_model, _ = _convert(cli_parser, params, True)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 524, in _convert
    raise e
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 476, in _convert
    ov_model = driver(argv, {"conversion_parameters": non_default_params})
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 226, in driver
    ov_model = moc_emit_ir(prepare_ir(argv), argv)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 172, in prepare_ir
    ov_model = moc_pipeline(argv, moc_front_end)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\moc_frontend\pipeline.py", line 247, in moc_pipeline
    ov_model = moc_front_end.convert(input_model)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\frontend\frontend.py", line 18, in convert
    converted_model = super().convert(model)
openvino._pyopenvino.OpConversionFailure: Check 'is_conversion_successful' failed at src\frontends\tensorflow\src\frontend.cpp:398:
FrontEnd API failed with OpConversionFailure:
[TensorFlow Frontend] Internal error, conversion is failed for FusedBatchNormV3 operation with a message:
Check 'TRShape::broadcast_merge_into(output_shape, input_shapes[1], autob)' failed at src\core\shape_inference\include\eltwise_shape_inference.hpp:28:
While validating node 'opset1::Subtract Subtract_50 (input_1[0]:f32[1,2,3,4,5], Unsqueeze_41[0]:f32[5,1,1,1]) -> (dynamic[...])' with friendly_name 'Subtract_50':
Argument shapes are inconsistent.
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by rkazants** [comment]
Hi @jikechao, I just tried to convert the model using 2023.2 release and both conversion from disk and memory works fine for me:

![image](https://github.com/openvinotoolkit/openvino/assets/35459624/a0caf875-8839-45d5-9742-bedeb520b6ad)

Also, I recently fixed 5D case for FusedBatchNorm and this fix should have gone to 2023.2 release: https://github.com/openvinotoolkit/openvino/pull/19904
Please check consistence of your environment.

Best regards,
Roman

**Comment by Liuqh12** [comment]
I have a similar error:

```python
import torch
import torch.nn

class Test(torch.nn.Module):
    def __init__(self, *args, **kwargs) -> None:
        super().__init__(*args, **kwargs)
        self.a = torch.rand((1, 64, 256, 256))
        self.b = torch.rand((1, 64, 256, 256))
        self.tail = torch.nn.Conv2d(in_channels=64, out_channels=1, kernel_size=3, stride =1, padding=1)
    
    def forward(self, args):  
        # can't convert      
        x = args * self.a        
        y = x * self.b
        return self.tail(y)
        
        # that is ok        
        # return self.tail(args)

if __name__ == '__main__':
    a = Test()
    a.eval()
    ins = torch.rand((1,64,256, 256))
    print(a(ins).shape)
    torch.onnx.export(a,  ins, "test.onnx", input_names=['input'], output_names=['output'])        
```

openvino version: 
```text
openvino           2024.2.0
openvino-telemetry 2024.1.0
```
Error:
```text
>>> import openvino as ov
>>> ov.convert_model('test.onnx')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\Miniforge3\envs\hed\lib\site-packages\openvino\tools\ovc\convert.py", line 100, in convert_model
    ov_model, _ = _convert(cli_parser, params, True)
  File "D:\Miniforge3\envs\hed\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 534, in _convert
    raise e
  File "D:\Miniforge3\envs\hed\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 480, in _convert
    ov_model = driver(argv, {"conversion_parameters": non_default_params})
  File "D:\Miniforge3\envs\hed\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 241, in driver
    ov_model = moc_emit_ir(prepare_ir(argv), argv)
  File "D:\Miniforge3\envs\hed\lib\site-packages\openvino\tools\ovc\moc_frontend\moc_emit_ir.py", line 20, in moc_emit_ir
    apply_moc_transformations(ngraph_function, cf=False, smart_reshape=True)
RuntimeError: Check 'TRShape::broadcast_merge_into(output_shape, input_shapes[1], autob)' failed at src/core/shape_inference/include\eltwise_shape_inference.hpp:28:
While validating node 'opset1::Multiply Multiply_20674 (opset1::Constant tail.weight[0]:f32[1,64,3,3], opset1::Constant Constant_20672[0]:f32[1,64,256,256]) -> (dynamic[...])' with friendly_name 'Multiply_20674':
Argument shapes are inconsistent.

>>>
```
